### PR TITLE
Treat InputIOExceptions during form submission as a record failure

### PR DESCRIPTION
**Pull request by amstone326**
fyi @ctsims

### CODE REVIEW

File: app/src/org/commcare/utils/FormUploadUtil.java [hunk 1]
```diff
@@ -169,9 +169,19 @@ private static FormUploadResult submitEntity(MultipartEntity entity, String url,
             return FormUploadResult.TRANSPORT_FAILURE;
         } catch (IOException | IllegalStateException e) {
             e.printStackTrace();
-            Logger.log(AndroidLogger.TYPE_ERROR_STORAGE,
-                    "Error reading form during submission: " + e.getMessage());
-            return FormUploadResult.TRANSPORT_FAILURE;
+            if (e instanceof InputIOException) {
```
Comment (ctsims) [inline app/src/org/commcare/utils/FormUploadUtil.java]: Is this caught above by the  } catch (InputIOException ioe) { line?

**Comment by ctsims** [comment]
Confirming that the issue appears to be that when we moved from writeFromInputToOutput to writeFromInputToOutputNew in StreamsUtil  (in submitEntity from FormUploadUtil), the new method didn't implement the directional stream exceptions, We can target 2.37 for that fix.

**Comment by amstone326** [comment]
@ctsims Can you open a ticket explaining what the issue is in a little bit more detail and assign it to the mobile backlog?

**Comment by ctsims** [comment]
done http://manage.dimagi.com/default.asp?255101

**Pull request closed without merging**